plugins/follow_me/follow_me_impl.cpp

```diff
@@ -0,0 +1,362 @@
+#include "follow_me_impl.h"
+#include "device_impl.h"
+#include "global_include.h"
+#include "px4_custom_mode.h"
+#include <functional>
+#include <cmath>
+
+namespace dronecore {
+
+using namespace std::placeholders; // for `_1`
+
+FollowMeImpl::FollowMeImpl() :
+    PluginImplBase(),
+    _curr_target_location {},
+    _time {},
+    _estimatation_capabilities(0),
+    _config {}
+{
+}
+
+FollowMeImpl::~FollowMeImpl()
+{
+}
+
+void FollowMeImpl::init()
+{
+    _parent->register_mavlink_message_handler(
+        MAVLINK_MSG_ID_HEARTBEAT,
+        std::bind(&FollowMeImpl::process_heartbeat, this, _1), static_cast<void *>(this));
+    set_default_config();
+}
+
+void FollowMeImpl::deinit()
+{
+    _parent->unregister_all_mavlink_message_handlers(this);
+}
+
+bool operator!(FollowMe::TargetLocation &location)
```

Comment:
What is this used for? It seems strange to me.

Reply:
:) I wanted to compare whether location was set or not by overloading `operator!`, I removed it now as it is not right way to compare real values. I am using `std::isfinite` as you suggested. Thank